src/group_theory/submonoid/basic.lean

```diff
@@ -79,6 +79,12 @@ namespace submonoid
 instance : set_like (submonoid M) M :=
 ⟨submonoid.carrier, λ p q h, by cases p; cases q; congr'⟩
 
+@[to_additive]
+def simps.coe (S : submonoid M) : set M := S
+
+initialize_simps_projections submonoid (carrier → coe)
+initialize_simps_projections add_submonoid (carrier → coe)
```

Comment:
@fpvandoorn, how hard would it be to teach `simp`s that `coe` should be a prefix?

Reply:
It shouldn't be too hard. It's on the list (#5489) and I'll try to implement it soon, since it has been requested multiple times now.